resources/task.rb

```diff
@@ -41,6 +43,7 @@
                                     :on_idle], :default => :hourly
 attribute :start_day, :kind_of => String, :default => nil
 attribute :start_time, :kind_of => String, :default => nil
+attribute :day, :kind_of => [ String, Integer ], :default => nil
```

Comment:
As a "correct by construction" fanatic, I wonder if this can be made cleaner/more explicit by having the frequency attribute take in objects for month and day that perform the appropriate validation.

I imagine something along the lines of:

```ruby

class Schedule

  def self.monthly_last_day(months=nil, start_day=nil, start_time=nil, end_date=nil)
    months = '*' if months.is_nil?
    Schedule.new(sc: 'MONTHLY', mo: 'LASTDAY', m: months, sd: start_date, ed: end_date, st: start_time)
  end

  def self.monthly(frequency=1, day=1, start_day=nil, start_time=nil, end_date=nil)
    raise "Only 12 months in a year" if frequency < 1 || frequency > 12
    raise "Only 31 days in a month" if day < 1 || day > 31
    Schedule.new(sc: 'MONTHLY', mo: frequency, d: day, sd: ...)
  end

  def self.monthly_on_week(week, week_day=nil, months=nil, start_day...)
      raise "Invalid week" unless [:first, :second, :third, :fourth, :last].includes?(week)
      validate_week_day week_day
      Schedule.new(sc: 'MONTHLY', mo: 'LAST', d: week_day, m: months, sd: ...)
  end

 ...

  # does ruby let you make this private?
  def initialize(sc, mo, m, d, sd, st, ed)
    # Validate m, sd, st, and ed
  end
end

attribute :schedule, :kind_of => [Schedule], :default => Schedule.hourly
```

This way, you can't actually write code in a way that gives you an inconsistent set of arguments.  You have to generate a schedule using one of the static generators and they force you to get it right.